Remove deprecated custom virtual environment feature

TheRealHaoLiu opened the pull request:
##### SUMMARY
Remove the deprecated custom virtual environment feature from AWX. This feature has been superseded by execution environments and is no longer needed. The removal includes all related models, API endpoints, management commands, configuration settings, and documentation references.

This change modernizes the codebase by eliminating legacy virtual environment management in favor of the standardized execution environment approach for job isolation and dependency management.

##### ISSUE TYPE
 - Breaking Change 

##### COMPONENT NAME
 - API
 - Collection
 - CLI
 - Docs

##### ADDITIONAL INFORMATION

**Background:**
The custom virtual environment feature was deprecated when execution environments were introduced as the preferred method for managing job dependencies and isolation. This PR completes the removal of all custom venv functionality.

**Changes Made:**
- **Models**: Removed `CustomVirtualEnvMixin` and `custom_virtualenv` fields from JobTemplate, Job, Project, Organization, InventorySource, and InventoryUpdate models
- **Management Commands**: Deleted `list_custom_venvs`, `export_custom_venv`, and `custom_venv_associations` commands
- **Configuration**: Removed `CUSTOM_VENV_PATHS` and `BASE_VENV_PATH` settings
- **API**: Removed `custom_virtualenv` fields from all serializers and API views
- **Collection**: Removed `custom_virtualenv` parameter from Ansible collection modules (job_template, project, inventory_source)
- **Utilities**: Removed `get_custom_venv_choices()` and `get_custom_venv_pip_freeze()` functions
- **Documentation**: Removed references from execution environments documentation
- **Database**: Added migration to remove `custom_virtualenv` columns from database tables

**Breaking Changes:**
- API endpoints will no longer return `custom_virtualenv` fields
- Ansible collection modules will no longer accept `custom_virtualenv` parameter
- Management commands for custom venv operations are no longer available
- Configuration settings `CUSTOM_VENV_PATHS` and `BASE_VENV_PATH` are removed

**Migration Path:**
Users should migrate to execution environments for dependency management. The existing execution environment functionality provides superior isolation and dependency management capabilities.

**Testing:**
- Verified all model references are removed
- Confirmed API serializers no longer include custom_virtualenv fields
- Validated collection modules work without custom_virtualenv parameter
- Database migration tested to ensure clean field removal

```bash
# Before: Custom venv fields were present in API responses
{
  "id": 1,
  "name": "My Job Template",
  "custom_virtualenv": "/path/to/venv",
  ...
}

# After: Custom venv fields are removed
{
  "id": 1,
  "name": "My Job Template",
  "execution_environment": 2,
  ...
}
```

TheRealHaoLiu committed:
Remove deprecated custom virtual environment feature

The custom virtual environment feature has been deprecated in favor of
execution environments. This commit completely removes all custom venv
functionality from the codebase.

Changes include:
- Remove CustomVirtualEnvMixin and custom_virtualenv fields from all models
- Remove custom venv management commands (list_custom_venvs, export_custom_venv, custom_venv_associations)
- Remove CUSTOM_VENV_PATHS and BASE_VENV_PATH configuration settings
- Remove custom_virtualenv from API serializers and views
- Remove custom_virtualenv parameter from Ansible collection modules
- Remove utility functions get_custom_venv_choices() and get_custom_venv_pip_freeze()
- Remove references from documentation and tests
- Add database migration to remove custom_virtualenv fields

This change is part of the modernization effort to standardize on
execution environments for job isolation and dependency management.

AlanCoding commented on docs/execution_environments.md:
```diff
@@ -49,16 +49,3 @@ Jobs will use the first available execution environment in this list:
 If more than one EE fits a criteria (applies for 6 and 7), then the most recently created one will be used.
 
 ## Migrating from Custom Virtual Environments
```
Should delete this line.

chrismeyersfsu replied:
remove

AlanCoding commented on awx/main/migrations/0202_remove_custom_virtualenv.py:
```diff
@@ -0,0 +1,37 @@
+            model_name='project',
+            name='custom_virtualenv',
+        ),
+    ]
```
I would like to merge this into 0201 migration

TheRealHaoLiu committed:
Fix unused import in root.py

Remove unused JobTemplate import that was left after removing custom venv functionality.

chrismeyersfsu commented:
https://github.com/ansible/awx/blob/31ee509dd5b8aae65effb65f4919cf33d9f391da/awx/main/access.py#L1593

https://github.com/ansible/awx/blob/31ee509dd5b8aae65effb65f4919cf33d9f391da/awx/api/metadata.py#L155

TheRealHaoLiu committed:
Remove custom_virtualenv references from core AWX and organization module

Clean up remaining custom_virtualenv references in 4 files:

- awx/api/metadata.py: Remove 'custom_virtualenv' from field type handling
- awx/main/access.py: Remove 'custom_virtualenv' from JobTemplate access fields
- awx/main/models/notifications.py: Remove from notification field lists and example data
- awx_collection/plugins/modules/organization.py: Remove parameter definition and usage

This continues the deprecation of the custom virtual environment feature
by removing references that were missed in previous cleanup efforts.
